File: dcp-solutions/2025-02-02/solution.js
```javascript
// Definition for singly-linked list.
class ListNode {
    val: number;
    next: ListNode | null;

    constructor(val = 0, next = null) {
        this.val = val;
        this.next = next;
    }
}

function swapPairs(head) {
    if (!head || !head.next) return head;

    const dummyNode = new ListNode(0);
    dummyNode.next = head;
    let prev = dummyNode;

    while (prev.next && prev.next.next) {
        const firstNode = prev.next;
        const secondNode = prev.next.next;

        // reverse the link of first node
        prev.next = secondNode;
        firstNode.next = secondNode.next;
        secondNode.next = firstNode;

        // move forward three nodes
        prev = firstNode;
    }

    return dummyNode.next;
}
```

File: dcp-solutions/2025-02-02/solution.go
```markdown
func swapPairs(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return head
    }
    
    newHead := head.Next
    head.Next = swapPairs(newHead.Next)
    newHead.Next = head
    
    return newHead
}
```

This solution uses a recursive approach to solve the problem. It first checks if the linked list has less than two nodes, in which case it just returns the head as there's nothing to swap. If the linked list has at least two nodes, it creates a new head by swapping the current node with the next node and then swaps the `next` pointer of the new head with the current node. The function then recursively calls itself on the rest of the linked list (i.e., the next node after the swapped pair).